### DISCUSSION TITLE
How can I retrieve the AI Assistant's result within a script?

### DISCUSSION THREAD

**Discussion by uglyboy-tl** [chhoumann/quickadd#759] [Q&A]
I am currently working with macro. Initially, I obtained a result using an AI Assistant. Subsequently, I intended to manipulate this result via a script. However, the `params.variables` array is empty. How should I retrieve this variable within the script?

Here is a part of the code:
```js
module.exports = async (params) => {
  const file = app.workspace.getActiveFile();
  let description = params.variables["description"];
  // 打印整个 params 对象以调试
  await app.fileManager.processFrontMatter(file, (fm) => {
    console.log("Params:", params);
    fm["description"] = `${description}`;
  });
};
```

**Comment by chhoumann** [comment]
Sorry for the slow reply. The result is there - it is just stored under a different name.

The AI Assistant writes its output to the variable named in the command's **Output variable name** setting (open the AI step's settings in the macro builder). The default is `output`, so either set it to `description`, or read `params.variables["output"]` in your script. The prompt template's name has no effect on the variable name.

Two things to double check:

- Both steps must be commands of the same macro - variables do not survive across separate runs.
- A quick `console.log(Object.keys(params.variables))` at the top of the script shows exactly what arrived.

Your `processFrontMatter` code is fine once the variable resolves (prefer `params.app` over the global `app` while you are at it).

https://quickadd.obsidian.guide/docs/AIAssistant
